### PR TITLE
Added unknown dependency finder changes in osioAnalysisFlow

### PR DESCRIPTION
**Pull request by samuzzal-choudhury**

### CODE REVIEW

**Comment by samuzzal-choudhury** [comment]
[test]

File: f8a_worker/dispatcher/flows/osioAnalysisFlow.yml [hunk 1]
```diff
@@ -4,5 +4,7 @@
       queue: '{DEPLOYMENT_PREFIX}_api_osioAnalysisFlow_v0'
       edges:
         - from:
-          to:
-            - 'dependency_tree'
\ No newline at end of file
+          to: 'GithubDependencyTreeTask'
+        - from: 'GithubDependencyTreeTask'
+          to: 'unknown_deps_fetcher'
```
Comment (miteshvp) [inline f8a_worker/dispatcher/flows/osioAnalysisFlow.yml]: It's better to have a uniformity in names here

File: f8a_worker/dispatcher/nodes.yml [hunk 2]
```diff
@@ -280,12 +280,18 @@
...
       queue: '{DEPLOYMENT_PREFIX}_{WORKER_ADMINISTRATION_REGION}_KeywordsSummaryTask_tagging_v0'
       storage: 'S3KeywordsSummary'
-    - name: 'dependency_tree'
+    - name: 'GithubDependencyTreeTask'
       classname: 'GithubDependencyTreeTask'
       import: 'f8a_worker.workers'
       max_retry: 0
       queue: '{DEPLOYMENT_PREFIX}_{WORKER_ADMINISTRATION_REGION}_GithubDependencyTreeTask_v0'
       storage: 'BayesianPostgres'
+    - name: 'unknown_deps_fetcher'
```
Comment (miteshvp) [inline f8a_worker/dispatcher/nodes.yml]: Uniform naming convention will definitely help

File: f8a_worker/workers/unknown_dep_fetcher.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+from f8a_worker.graphutils import GREMLIN_SERVER_URL_REST
+from f8a_worker.base import BaseTask
+from f8a_worker.utils import get_session_retry
+
+class UnknownDependencyFetcherTask(BaseTask):
+    _analysis_name = 'unknown_deps_fetcher'
+    description = 'Fetch unknown dependencies'
+
+    def get_dependency_data(self, dependency_list):
+        ecosystem = "maven"
```
Comment (miteshvp) [inline f8a_worker/workers/unknown_dep_fetcher.py]: do we want this hard-coding?

File: f8a_worker/dispatcher/flows/osioAnalysisFlow.yml (hunk 1 above)
Comment (samuzzal-choudhury) [reply f8a_worker/dispatcher/flows/osioAnalysisFlow.yml]: @miteshvp Similar convention was followed in earstwhile stackAnalysisV2Flow as well, but I will surely make this uniform in an uncoming PR after the complete flow is implemented.

File: f8a_worker/dispatcher/nodes.yml (hunk 2 above)
Comment (samuzzal-choudhury) [reply f8a_worker/dispatcher/nodes.yml]: @miteshvp Similar convention was followed in erstwhile stackAnalysisV2Flow as well, but I will surely make this uniform in an upcoming PR after the complete flow is implemented.

File: f8a_worker/workers/unknown_dep_fetcher.py (hunk 3 above)
Comment (samuzzal-choudhury) [reply f8a_worker/workers/unknown_dep_fetcher.py]: @miteshvp The parent GithubDependencyTreeTask implementation was done by the team assuming maven ecosystem at this moment. In order to change this, the parent implementation has to be generalized. We will implement this when we are going to add support for other ecosystems in sprint 146.

**Comment by samuzzal-choudhury** [comment]
@miteshvp thanks for reviewing and all the valid review comments. Can we take those changes up as part of an upcoming PR? The current priority is to set up the entire flow in place.

**Comment by miteshvp** [comment]
Please merge it but let's document this somewhere

**Pull request merged**